build_tools/build/src/ci_gen/job.rs

```diff
@@ -262,6 +263,8 @@ impl JobArchetype for JvmTests {
                     step::engine_test_reporter(target, graal_edition),
                 ]
             })
+            // Clean always, because JVM tests are unstable.
+            .cleaning(CleaningCondition::Always)
```

Comment:
> Always cleaning in JVM tests job is required as per @Akirathan.

I think you got the wrong idea. Let me clarify.  It is not always required, I am personally just doing it because it usually makes sense because of the nature of my PRs. However, always cleaning shouldn't be necessary, for example for library guys. Remove it please.